core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -253,27 +255,32 @@ private void validate(SnapshotRequest request, ClusterState state) {
         if (repositoriesMetaData == null || repositoriesMetaData.repository(request.repository()) == null) {
             throw new RepositoryMissingException(request.repository());
         }
-        if (!Strings.hasLength(request.name())) {
-            throw new InvalidSnapshotNameException(new SnapshotId(request.repository(), request.name()), "cannot be empty");
+        validate(new SnapshotId(request.repository(), request.name()));
+    }
+    
+    private static void validate(SnapshotId snapshotId) {
+        if (!Strings.hasLength(snapshotId.getSnapshot())) {
+            throw new InvalidSnapshotNameException(snapshotId, "cannot be empty");
         }
```

Comment:
can we assign `snapshotId.getSnapshot()` to a local var instead of calling it over and over again?